src/aggregate/expr/parser.y

```diff
@@ -57,40 +59,88 @@
 program ::= expr(A). { ctx->root = A; }
 
 expr(A) ::= LP expr(B) RP. { A = B; }
-expr(A) ::= expr(B) PLUS expr(C). { A = RS_NewOp('+', B, C); }
-expr(A) ::= expr(B) DIVIDE expr(C). {  A = RS_NewOp('/', B, C); }
-expr(A) ::= expr(B) TIMES expr(C). {  A = RS_NewOp('*', B, C);}
-expr(A) ::= expr(B) MINUS expr(C). {  A = RS_NewOp('-', B, C); }
-expr(A) ::= expr(B) POW expr(C). {  A = RS_NewOp('^', B, C); }
-expr(A) ::= expr(B) MOD expr(C). { A = RS_NewOp('%', B, C); }
+
+// "Manual" expansion of the arithmetic operators, to optimize the AST in-place when possible
+// expr ::= expr OP expr
+expr(A) ::= expr(B) PLUS   expr(C). { A = RS_NewOp('+', B, C); }
+expr(A) ::= expr(B) DIVIDE expr(C). { A = RS_NewOp('/', B, C); }
+expr(A) ::= expr(B) TIMES  expr(C). { A = RS_NewOp('*', B, C); }
+expr(A) ::= expr(B) MINUS  expr(C). { A = RS_NewOp('-', B, C); }
+expr(A) ::= expr(B) POW    expr(C). { A = RS_NewOp('^', B, C); }
+expr(A) ::= expr(B) MOD    expr(C). { A = RS_NewOp('%', B, C); }
+// expr ::= number OP expr
+expr(A) ::= number(B) PLUS   expr(C). { A = RS_NewOp('+', RS_NewNumberLiteral(B), C); }
+expr(A) ::= number(B) DIVIDE expr(C). { A = RS_NewOp('/', RS_NewNumberLiteral(B), C); }
+expr(A) ::= number(B) TIMES  expr(C). { A = RS_NewOp('*', RS_NewNumberLiteral(B), C); }
+expr(A) ::= number(B) MINUS  expr(C). { A = RS_NewOp('-', RS_NewNumberLiteral(B), C); }
+expr(A) ::= number(B) POW    expr(C). { A = RS_NewOp('^', RS_NewNumberLiteral(B), C); }
+expr(A) ::= number(B) MOD    expr(C). { A = RS_NewOp('%', RS_NewNumberLiteral(B), C); }
```

Comment:
Is there an added value for these rules other than saving us a reduce and a shift?

Reply:
YES

1. the `number OP expr` rule is required for the cases we shifted on the `number` before and we cannot reduce it now that we have advanced to the right `expr`
2. the `expr OP number` rule is required so the priority of the rule is of the **operation** and not the priority of the reducing of `numeric` to `expr`. Otherwise, queries like `@n * 2 + 1` may be interpreted as `@n * 3`

Reply:
Two things:
1. The first number that is shifted becomes an `expr` due to the rule `expr(A) ::= number(B)`.
Then, an `OP` arrives and is shifted.
Then, another number is shifted.
If no other rule matches the stack, it will become an `expr`, and then the `expr OP expr` will match.
**Then**, the next number (say `1`) will be shifted with the same logic.
2. How can the `number OP expr` happen? Wouldn't the number would become an `expr` upon the first shift-reduce?

Reply:
OK, makes sense 👍 
Two notes:
1. Consider adding the computation of the `number OP number` in the `expr OP expr` rule code such that some rule overhead and precedence complexity is saved.
2. If we stay with this instead of incorporating (1), let's have some note here explaining why these rules are needed, or point to some test that explains it.